### DISCUSSION TITLE
Private shared preset cannot be loaded from public github repo within the same org

### DISCUSSION THREAD

**Discussion by jetersen** [renovatebot/renovate#9002] [Request Help]
**What Renovate type, platform and version are you using?**

hosted app

**Describe the bug**

We have public repo that attempts to use a shared presets on a private repo:
https://github.com/Specshell/specshell.software.maven.pom/blob/main/.github/renovate.json
https://github.com/Specshell/specshell.software.maven.pom/issues/59
We have several other private repos that successfully can retrieve the same config.
I guess renovate on a public repo is not correctly using the github app token provided.

![image](https://user-images.githubusercontent.com/1661688/110196703-90f94d80-7e46-11eb-932e-0b76b44c8db1.png)

**Relevant debug logs**

```
DEBUG: migrated config
{
  "config": {
    "$schema": "https://docs.renovatebot.com/renovate-schema.json",
    "extends": [
      "github>whitesource/merge-confidence:beta",
      "github>Specshell/renovate-config"
    ]
  }
}
DEBUG: GitHub 404
{
  "url": "https://api.github.com/repos/Specshell/renovate-config/contents/default.json"
}
DEBUG: Failed to retrieve default.json from repo
{
  "statusCode": 404,
  "url": "https://api.github.com/repos/Specshell/renovate-config/contents/default.json"
}
DEBUG: default.json preset not found - trying renovate.json
DEBUG: GitHub 404
{
  "url": "https://api.github.com/repos/Specshell/renovate-config/contents/renovate.json"
}
DEBUG: Failed to retrieve renovate.json from repo
{
  "statusCode": 404,
  "url": "https://api.github.com/repos/Specshell/renovate-config/contents/renovate.json"
}
DEBUG: Preset fetch error
{
  "preset": "github>Specshell/renovate-config",
  "err": {
    "message": "dep not found",
    "stack": "Error: dep not found\n    at fetchJSONFile (/home/ubuntu/renovateapp/node_modules/renovate/dist/config/presets/github/index.js:22:15)\n    at runMicrotasks (<anonymous>)\n    at processTicksAndRejections (internal/process/task_queues.js:93:5)\n    at async Object.fetchPreset (/home/ubuntu/renovateapp/node_modules/renovate/dist/config/presets/util.js:22:27)\n    at async getPreset (/home/ubuntu/renovateapp/node_modules/renovate/dist/config/presets/index.js:166:24)\n    at async Object.resolveConfigPresets (/home/ubuntu/renovateapp/node_modules/renovate/dist/config/presets/index.js:227:37)\n    at async mergeRenovateConfig (/home/ubuntu/renovateapp/node_modules/renovate/dist/workers/repository/init/config.js:194:52)\n    at async Object.getRepoConfig (/home/ubuntu/renovateapp/node_modules/renovate/dist/workers/repository/init/config.js:230:14)\n    at async Object.initRepo (/home/ubuntu/renovateapp/node_modules/renovate/dist/workers/repository/init/index.js:19:14)\n    at async Object.renovateRepository (/home/ubuntu/renovateapp/node_modules/renovate/dist/workers/repository/index.js:62:18)\n    at async renovateRepository (/home/ubuntu/renovateapp/app/worker/index.js:364:26)\n    at async /home/ubuntu/renovateapp/app/worker/index.js:597:5"
  }
}
INFO: Throwing preset error
{
  "validationError": "Cannot find preset's package (github>Specshell/renovate-config)"
}
```

**Have you created a minimal reproduction repository?**

Please read the [minimal reproductions documentation](https://github.com/renovatebot/renovate/blob/master/docs/development/minimal-reproductions.md) to learn how to make a good minimal reproduction repository.

- [ ] This is a really small bug, it does not need a reproduction (think small typo)
- [ ] I have provided a minimal reproduction repository
- [ ] I don't have time for that, but it happens in a public repository I have linked to
- [ ] I don't have time for that, and cannot share my private repository
- [x] The nature of this bug means it's impossible to reproduce publicly

**Additional context**

**Comment by rarkins** [comment]
Public repos receive a token which is only valid for the current repo and does not grant private access to others.

**Reply by bigpopakap** [reply]
Hello! Good to know. Is there any suggested workaround for this?

**Reply by viceice** [reply]
use public presets

**Reply by bigpopakap** [reply]
That makes total sense probably for most use cases. Use a public config for public repositories, and only use a private config for private repositories that need access tokens.

I should clarify my use case: I have a public repository that uses a private node dependency XYZ from a private GitHub repo. (This is fine because I don't mind publicly exposing the _existence_ of XYZ, I just don't want anyone to be able to actually install XYZ if they clone the repo). So I'd need some access token in my config to allow Renovate to access XYZ, but that would have to be in a private config.

Is there any workaround to support _that_ use case?

**Reply by rarkins** [reply]
A private node dependency is not the same as a private preset. Please create a separate discussion and be more specific about your example.
